### ISSUE TITLE
`ssh issue` seems unusable on Windows

### ISSUE THREAD

**Issue by mrkishi** [superfly/flyctl#944]
It doesn't seem possible to access issued SSH credentials on Windows.

I've tried setting them up both with Windows' supplied `ssh-agent` and (separately) with the latest build of Microsoft's [Win32-OpenSSH](https://github.com/powershell/Win32-OpenSSH), unsuccessfully:
```
> fly ssh issue org email --agent
Error can't connect to SSH agent: dial unix: missing address
```

And I can't export the key file either:
```
> fly ssh issue org email key --dotssh
***Can't open private key file: open /.ssh/key: The system cannot find the path specified.
***Can't open private key file: open /.ssh/key: The system cannot find the path specified.
***Can't open private key file: open /.ssh/key: The system cannot find the path specified.
***Can't open private key file: open /.ssh/key: The system cannot find the path specified.
***Can't open private key file: open /.ssh/key: The system cannot find the path specified.
...
```

While `fly ssh console` works, it won't handle all use-cases (eg. is there a way to `scp` through it?).

### System information
- OS: `Windows 10` `build 19044.1645`
- Fly: `flyctl.exe v0.0.321` `windows/amd64` 1c1d276
- System OpenSSH: `OpenSSH_for_Windows_8.1p1, LibreSSL 3.0.2`
- User OpenSSH: `OpenSSH_for_Windows_8.9p1, LibreSSL 3.4.3`

**Comment by mrkishi** [comment]
Looking through the source I found a simple workaround for anyone facing the same issue (pwsh syntax):

```sh
> $env:HOME = 'x:/some/path'
> mkdir x:/some/path/.ssh -f
> fly ssh issue --dotssh
...
Wrote 24-hour SSH credential to x:/some/path/.ssh/key, x:/some/path/.ssh/key-cert.pub
```

**Comment by jsierles** [comment]
In #949, things seem to go further as `SSH_AUTH_SOCK` is set but fails to connect. Is there more than one SSH service in play here?

**Comment by mrkishi** [comment]
Since @lyj0309 executed `start-ssh-agent.cmd` I assume they're using Git for Windows's OpenSSH, which I _believe_ runs on the MSYS2 POSIX emulation layer.

I think MSYS2's and Microsoft's OpenSSH operate slightly differently, with Microsoft's `ssh-agent` [using a static IPC port](https://github.com/PowerShell/Win32-OpenSSH/wiki/About-Win32-OpenSSH-and-Design-Details) instead of `SSH_AUTH_SOCK`, for instance. Both provide a working `ssh-add`, though, so perhaps that could be a way to get it working without a deep dive on OpenSSH behavior on Windows.

**Comment by darwin67** [comment]
Looks like it doesn't work on Linux too.

```
❯ fly ssh issue --agent
? Select Organization: <org>
Error can't connect to SSH agent: dial unix: missing address
```

System spec:
```
❯ neofetch
                   -`                    Framework
                  .o+`                   ----------------
                 `ooo/                   OS: Arch Linux x86_64
                `+oooo:                  Host: Laptop AB
               `+oooooo:                 Kernel: 6.1.8-arch1-1
               -+oooooo+:                Uptime: 5 days, 6 hours, 5 mins
             `/:-:++oooo+:               Packages: 1397 (pacman), 6 (flatpak)
            `/++++/+++++++:              Shell: zsh 5.9
           `/++++++++++++++:             Resolution: 2256x1504
          `/+++ooooooooooooo/`           WM: sway
         ./ooosssso++osssssso+`          Theme: Adwaita [GTK2]
        .oossssso-````/ossssss+`         Icons: Adwaita [GTK2]
       -osssssso.      :ssssssso.        Terminal: tmux
      :osssssss/        osssso+++.       CPU: 11th Gen Intel i7-1165G7 (8) @ 4.700GHz
     /ossssssss/        +ssssooo/-       GPU: Intel TigerLake-LP GT2 [Iris Xe Graphics]
   `/ossssso+/:-        -:/+osssso+-     Memory: 11242MiB / 31888MiB
  `+sso+:-`                 `.-/+oso:
 `++:.                           `-/+/
 .`                                 `/
```

**Comment by redjonzaci** [comment]
Do we still need this?

**Comment by mrkishi** [comment]
Since it now defaults to outputting the keys wherever you point it to, at least the main feature works without workarounds.

- `fly ssh issue`: successfully generates the keys
- `fly ssh issue --agent`: `Error: can't connect to SSH agent: dial unix: missing address`
- `fly ssh issue --dotssh`: `Can't open private key file: open /.ssh/keyname: The system cannot find the path specified.`

@redjonzaci I'm going to leave it open so you folks can decide whether these erroring extra options are fine as-is or deserve to be documented/removed/fixed on Windows.

**Comment by redjonzaci** [comment]
Thanks for the response!
Will take a look at those in a few hours.

**Comment by redjonzaci** [comment]
`fly ssh issue --dotssh` expects that `.ssh` folder exists.
I wonder if we should create it if it doesn't.
Should we also update the path where `.ssh` folder is expected based on OS?
Like, in Windows, `ssh-keygen` creates it under `C:\Users\{currentUser}\.ssh`.

**Comment by redjonzaci** [comment]
I meant in Windows, we could try to create it in `C:\Users\{currentUser}\.ssh\`, not in `/.ssh/`.

As for the note, `fly ssh issue` works with `--org` flag instead of `org` argument as of [v0.1.85](https://github.com/superfly/flyctl/releases/tag/v0.1.85).
And it will err if you provide arguments to it when https://github.com/superfly/flyctl/pull/2777 is merged
as we are switching all its arguments to flags.

**Comment by mrkishi** [comment]
Ah, I see. In that case the command help's usage section is misleading:

```sh
❯ fly version
flyctl.exe v0.1.90 windows/amd64 Commit: c7497d40cc85d887dfd149b9cac3c675ab624e5a BuildDate: 2023-09-07T17:33:12Z

❯ fly help ssh issue
Issue a new SSH credential. With -agent, populate credential into SSH agent. With -hour, set the number of hours (1-72)
for credential validity.

Usage:
  flyctl ssh issue [org] [path] [flags]

[...]
```

**Comment by redjonzaci** [comment]
@mrkishi yeah,the PR that changes those args to flags hasn't been merged yet.
